streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -157,6 +159,7 @@ public void resume() {
         log.debug("{} Resuming", logPrefix);
         if (eosEnabled) {
             producer.beginTransaction();
+            transactionInFlight = true;
```

Comment:
we seem to be introducing booleans all over the place. The more we add the harder it gets to follow the logic as we have `if(someflag)` all over the place. Generally this is pretty poor OO design.
Not picking on this one particularly, just in general